sklearn/ensemble/_forest.py

```diff
@@ -679,6 +682,33 @@ def feature_importances_(self):
         all_importances = np.mean(all_importances, axis=0, dtype=np.float64)
         return all_importances / np.sum(all_importances)
 
+    def _get_estimators_indices(self):
```

Comment:
Does this mainly apply in the context of when bootstrap=True?

Reply:
No, the `_n_samples_bootstrap` is used to determine if bootstrapping is done or not and to what degree.

This is a similar private function to the Bagging ensemble: https://github.com/scikit-learn/scikit-learn/blob/acf60def1cff9d440c86b34e9a27a8c44de4d71f/sklearn/ensemble/_bagging.py#L512

Reply:
I added an instance of the unit-test to show that `bootstrap=False` works too. The main goal is to allow the user to introspect which samples of training data X was used to fit a specific instance of the decision tree

Reply:
I think when bootstrap is False then n_samples_bootstrap is equal to the number of samples self._n_samples.
When _generate_sample_indices is called with these values it produces a random set of indices which can contain duplicates as well. For example the len(sample_indices) == 200 but len(set(sample_indices)) == 130. Do we want this behaviour?

Reply:
Thanks!